packages/flutter/lib/src/cupertino/scrollbar.dart

```diff
@@ -183,6 +189,25 @@ class _CupertinoScrollbarState extends State<CupertinoScrollbar> with TickerProv
         ..color = CupertinoDynamicColor.resolve(_kScrollbarColor, context)
         ..padding = MediaQuery.of(context).padding;
     }
+    WidgetsBinding.instance.addPostFrameCallback((duration) {
+      if (widget.displayAlways) {
+        assert(widget.controller != null);
+        widget.controller.position.didUpdateScrollPositionBy(0);
+      }
+    });
```

Comment:
Also, I think your logic is missing a few edge cases now.  We need to make sure all of these cases work:

  1. On first render with `displayAlways: true`, the thumb shows.
  1. On first render with `displayAlways: false`, the thumb is hidden.
  1. With `displayAlways: true`, fling a scroll. While it's still scrolling, set `displayAlways: false`. The thumb should not fade out until the scrolling stops.
  1. With `displayAlways: false`, fling a scroll. While it's still scrolling, set `displayAlways: true`. The thumb should not fade even after the scrolling stops.
  1. Toggling `displayAlways` while not scrolling fades the thumb in/out. This works even when you have never scrolled at all yet.

Let me know if I'm forgetting any edge cases.

Reply:
You may need to call `didUpdateScrollPositionBy` even when `displayAlways` is false, so that you can handle case where you toggle `displayAlways` to true after it started out as false, without ever scrolling.